providers/catalog.go

```diff
@@ -447,4 +448,23 @@ var catalog = CatalogType{ // nolint:gochecknoglobals
 			Write:     false,
 		},
 	},
+
+	// Microsoft Dynamics 365 Business Central configuration
+	MicrosoftDynamics365BusinessCentral: {
+		AuthType: Oauth2,
+		BaseURL:  "https://api.businesscentral.dynamics.com",
+		OauthOpts: OauthOpts{
+			AuthURL:                   "https://login.microsoftonline.com/{{.workspace}}/oauth2/v2.0/authorize",
```

Comment:
Can you also add `TokenMetadataFields` for this? You will also need to rebase this PR.